### PR TITLE
Publish own artifact to jitpack

### PR DESCRIPTION
**Pull request by axthosarouris**
Originally wanted to fix wrong Java version choice in Jitpack, but ended up building the artifacts in a controlled and repricable way and publishing them in Jitpack. 

Publishing to Jitpack is not meant as a replacement for Maven Central. It is meant to use during development

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -112,3 +119,41 @@ buildConfig {
...
+                    fromResolutionResult()
+                }
+            }
+            pom {
+                name = project.name
+                description = 'Java implementation of NVA Datamodel'
+                url = 'https://github.com/BIBSYSDEV/nva-datamodel-java'
+                licenses {
+                    license {
+                        name = 'The Apache License, Version 2.0'
```
Comment (janbj) [inline build.gradle:142]: Is this the right license?

**Comment by mattiahj** [comment]
Why not use the sonatype snapshot repository? I don't think we should add a second source for our artifacts.

**Review by brinxmat — APPROVED** [review]

**Review by janbj — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -6,12 +6,14 @@ plugins {
     id 'nebula.lint' version '16.8.0'
```
Comment (mattiahj) [inline build.gradle:6]: Why not use the sonatype snapshot repository? I don't think we should add a second source for our artifacts.

**Review by mattiahj — DISMISSED** [review]
Why not use the sonatype snapshot repository? I don't think we should add a second source for our artifacts, and a second way of publishing them.

**Comment by axthosarouris** [comment]
The snapshot would require merging something to main. We want to use Jitpack when we are making changes to two repos at the same time and we want to use the git hash value as a version number. It is a basically a tool for using versions that are not mature for publishing or even merging to main. This allows to test if things build remotely, e.g. the PR is green or your code runs as expected in the Cloud.

I agree that it is a bit of a hassle to have two artifactories, but the aim of having Jitpack is to be able to test some code that exists in one module in another module.  If there is another solution that involves only maven central that would be awesome.

**Comment by mattiahj** [comment]
> The snapshot would require merging something to main. 

I don't think so? I mean, we could have a separate Github Action for snapshots, or use a single one for publishing both releases and snapshots. SNAPSHOTS are by definition usually built from stuff merged to develop.

**Pull request merged**